tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -0,0 +1,233 @@
+        'manager': os.path.join(configurations_dir, 'manager.yaml'),
+        'agent': os.path.join(configurations_dir, 'agent.yaml')
+        }
+
+
+def load_vulnerability_detector_configurations():
+    """Return the configurations for Vulnerability testing for the agent and manager roles
+
+    Return:
+        Dict: Configurations for each role
+    """
+    return {
+        'agent': load_configuration_template(configurations_paths['agent'], [{}], [{}]),
+        'manager': load_configuration_template(configurations_paths['manager'], [{}], [{}])
+    }
+
+
+@pytest.fixture(scope='module')
+def setup_vulnerability_tests(host_manager: HostManager) -> Generator:
+    """Setup the vulnerability tests environment
+
+    Args: 
+        host_manager (HostManager): An instance of the HostManager class.
+    """
+    logger.error("Init setup of environment")
+
+    # Configure managers and agents
+    logger.error("Getting backup of current configurations")
+    hosts_configuration_backup = backup_configurations(host_manager)
+    logger.error("Configuring environment")
+    configure_environment(host_manager, load_vulnerability_detector_configurations())
+
+    # Restart managers and stop agents
+    logger.error("Stopping agents")
+    host_manager.control_environment('stop', ['agent'])
+    logger.error("Restarting managers")
+    host_manager.control_environment('restart', ['manager'])
+
+    # Wait until VD is updated
+    logger.error("Wait until Vulnerability Detector has update all the feeds")
+    wait_until_vd_is_updated(host_manager)
+
+    # Truncate alerts and logs of managers and agents
+    logger.error("Truncate managers and agents logs")
+    truncate_logs(host_manager)
+
+    # Start agents
+    logger.error("Starting agents")
+    host_manager.control_environment('start', ['agent'])
+
+    yield
+
+    logger.error("Restoring original configuration")
+    restore_backup(host_manager, hosts_configuration_backup)
+
+
+def check_vuln_state_index(host_manager: HostManager, agents_vulnerabilities_api_value: Dict[str, List[Dict]]):
+    """Check the consistency of the vulnerabilities included in the indexer.
+
+    This function ensures that the vulnerabilities gathered from the Wazuh API are included in the
+    Agent's Vulnerabilities index.
+
+    Args:
+        host_manager (HostManager): Host manager of the environment.
+        agents_vulnerabilities_api_value (Dict[str, List[str]]): A dictionary containing the list of vulnerabilities
+            for each agent, where keys are agent names and values are lists of strings representing the vulnerabilities.
+
+    Raises:
+        AssertionError: If inconsistencies are found between the API values and the state of the indexer.
+    """
+    # Get the indexer values
+    state_index_content_before_second_scan = get_indexer_values(host_manager)
+
+    # To Do: Ensure consistency of the agent_vulnerabilities_api_value and state_index_content_before_second_scan
+    pass
```

Comment:
`check_vuln_state_index` has been already created, is this expected?

Reply:
Done b8212e32a52be121af9f6e3bc76b1a6df3803bb6